[Feature] [SWCK] Add Instrument CRD to config and operate instruments. (#66)
apache/skywalking-swck: docs/java-agent-injector.md

The java agent injector supports a precedence order to configure the agent:

> ``` Annotations > SwAgent CR > Configmap > Default configmap```

Reviewer: I highly recommend to remove the namespaced configmap which can be substituted by `SwAgent`. If we though this pruning will affect too many pieces, a `deprecated` label should be attached to it.

Author: what do you think about:

``` Annotations > SwAgent > default config ```

of 

``` Annotations > SwAgent CR > Configmap (deprecated) > default config```

Author: I think removing agent.config configmap would cause many changes. i don't know if simply removing the `skywalking-swck-java-agent-configmap` volume would work. it looks like that we should also add the default file into agent image.  Maybe we can open another issue to discuss about it.

### Annotations

Annotations are described in [kubernetes annotations doc](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/).

We support annotations in [agent annotations](#Use-annotations-to-overlay-default-agent-configuration) and [sidecar annotations](#configure-sidecar).

> ### SwAgent CR

Reviewer: Please remove `CR` here. We don't have to point out it's a custom resource.

> Customer Resources are described in [kubernetes resources doc](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/).

Reviewer: ```suggestion
SwAgent is a [Customer Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) defined by SWCK.
```

    name: java-agent-configmap-volume
```

### Use SwAgent CR to overlay default agent configuration

The injector will read the SwAgent CR when pods creating.

SwAgent CRD basic structure is like:

```yaml
apiVersion: operator.skywalking.apache.org/v1alpha1
kind: SwAgent
metadata:
  name: swagent
  namespace: skywalking-system
spec:

>   containerMatcher: '.*'

Reviewer: The default value should be empty. If users don't specific this field, all containers would get injected.

  selector:
    app: demo
  javaSidecar:
    name: swagent
    image: apache/skywalking-java-agent:8.8.0-java8
    env:
      - name: SW_LOGGING_LEVEL
        value: "DEBUG"

>   sharedVolume:

Reviewer: What will the `sharedVolume` do? Could you elaborate it in this doc?

Author: Sure

    name: "sky-agent"
    mountPath: "/sky/agent"

>   swConfigMapVolume:

Reviewer: From the `description`, this refers to the `default` configmap. Why do users have to set up it since it's the deault?
